pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -73,14 +78,25 @@ type operationGenerator struct {
 }
 
 // NewOperationGenerator is returns instance of operationGenerator
-func NewOperationGenerator(kubeClient clientset.Interface,
+func NewOperationGenerator(
+	kubeResourcePartitionClient clientset.Interface,
 	volumePluginMgr *volume.VolumePluginMgr,
 	recorder record.EventRecorder,
 	checkNodeCapabilitiesBeforeMount bool,
-	blkUtil volumepathhandler.BlockVolumePathHandler) OperationGenerator {
+	blkUtil volumepathhandler.BlockVolumePathHandler,
+	kubeTenantPartitionClients ...[]clientset.Interface) OperationGenerator {
+
+	// kubeTenantPartitionClients is for the case when NewOperationGenerator is called from kubelet
+	var tenantClients []clientset.Interface
+	if len(kubeTenantPartitionClients) > 0 {
+		tenantClients = kubeTenantPartitionClients[0] // call is from kubelet
```

Comment:
Guess we need to calculate the corresponding TP client from the tenant, instead of using TP1?

Reply:
this we need further discussion on the design. will be address later.

Reply:
OK. Please consider it resolved.